### PR TITLE
feat(ST-123): integrate the file upload endpoint for resources

### PR DESCRIPTION
**Pull request by akib-raihan-sazim**
## Description of Change
- Updated ClassworkContainer and CreateFileButton so that it can accept classroom as a props
- Updated MaterialFormModal so that it can accept classroomId and call api 
- Created Classworks.api.ts file and integrated the endpoint
- Created interface for all the components

## Associated Ticket Link(s)
- https://trello.com/c/yeJGKyts

## Screenshots / Screen Recordings

https://github.com/user-attachments/assets/0d2c4af1-1e0c-4779-8b7e-bf75db66cae2

Screen Shot 2024-09-19 at 12 54 39 AM

### CODE REVIEW

**Review by ragib-hossain-sazim — APPROVED** [review]

File: modules/Classroom/components/MaterialFormModal/MaterialFormModal.tsx [hunk 1]
```diff
@@ -31,16 +35,35 @@ const MaterialFormModal: React.FC<IMaterialFormModalProps> = ({ opened, onClose
...
+        color: "blue",
+      });
+
       reset();
       onClose();
-    }, 1000);
+    } catch (error) {
+      showNotification({
+        title: "Upload failed",
+        message: (error as ApiError).data?.message,
```
Comment (ishmam-iqbal-sazim) [inline modules/Classroom/components/MaterialFormModal/MaterialFormModal.tsx:61]: Isn't there a parseApiErrorMessage method to structure the error messages?

File: modules/Classroom/containers/ClassworkContainer/ClassworkContainer.tsx [hunk 2]
```diff
@@ -1,10 +1,11 @@
 import { Box, Divider, Title } from "@mantine/core";
 
 import CreateFileButton from "../../components/CreateFileButton/CreateFileButton";
+import { IClassworkContainerProps } from "./ClassworkContainer.interface";
 
-const ClassworkContainer = () => (
+const ClassworkContainer: React.FC<IClassworkContainerProps> = ({ classroom }) => (
   <Box py="sm" px={{ base: 0, sm: "xl" }}>
-    <CreateFileButton />
+    <CreateFileButton classroom={classroom} />
```
Comment (ishmam-iqbal-sazim) [inline modules/Classroom/containers/ClassworkContainer/ClassworkContainer.tsx:8]: Do you need to pass the classroom object or only the id?

File: shared/redux/rtk-apis/classworks/classworks.api.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  endpoints: (builder) => ({
+    uploadResource: builder.mutation<IClassworkResource, CreateResourceDto>({
+      query: ({ file, classroomId, title, description }) => {
+        const formData = new FormData();
+        formData.append("file", file);
+        formData.append("title", title);
+        formData.append("description", description);
+
+        return {
+          url: `/classworks/${classroomId}/resources`,
```
Comment (ishmam-iqbal-sazim) [inline shared/redux/rtk-apis/classworks/classworks.api.ts:14]: Let's follow REST principles

```suggestion
          url: `/classrooms/${classroomId}/resources`,
```

**Review by ishmam-iqbal-sazim — APPROVED** [review]
Approved pending comments

File: modules/Classroom/components/MaterialFormModal/MaterialFormModal.tsx (hunk 1 above)
Comment (akib-raihan-sazim) [reply modules/Classroom/components/MaterialFormModal/MaterialFormModal.tsx:61]: Not sure, I have created this ApiError earlier in the project so I am using that

**Pull request merged**